### PR TITLE
PLT-777 Remove plutus ledger from Marconi' dependencies

### PR DESCRIPTION
**Pull request by berewt**
Here are the major type changes:

* `TxOutRef` (from `plutus-ledger-api`) has been replaced by an alias with the same name to `TxIn` of `cardano-api` that conveys the same information.
* `Address` (from `plutus-ledger`) has been replaced by `AddressAny` from `cardano-api`, that is basically an address in era, without the `era` witness type.
* `TxOut` (from `plutus-ledger`) has been replaced by `TxOut` from `cardana-api`.
* `Datum` and `DatumHash` (from `plutus-ledger-api`) has been replaced by `ScriptData` and `Hash ScriptData` from `cardano-api`.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [ ] Reviewer requested

### CODE REVIEW

**Comment by berewt** [comment]
I'm gonna make a PR to cardano-nod to expose the function I needed to duplicate and fix the corresponding comment in my code once before merging it.

**Comment by koslambrou** [comment]
You're waiting for the node PR to merge before merging this PR?

**Comment by berewt** [comment]
> You're waiting for the node PR to merge before merging this PR?

I think we'd better merge it before the node PR, just to avoid adding new features that rely on `plutus-ledger` in the meantime.

**Review by kayvank — APPROVED** [review]
Thank you for doing the work. I specially like :heart: the Marconi.Types and plan on moving some of data structure to there from the Indexers.hs

**Comment by berewt** [comment]
> I'll merge mine after this PR is merged to main

If yours is ready to merge, don't worry and merge it, I'll handle the conflict if needed.

**Review by raduom — APPROVED** [review]
LGTM

File: marconi/src/Marconi/Index/Utxo.hs
Comment (eyeinsky) [inline marconi/src/Marconi/Index/Utxo.hs:188]: I think here the original version was clearer. I don't know what `has` and `only` do so to know it I'd have to go and look at lens documentation and even then it will take a bit. The original version on the other hand pretty much says what it does.

File: marconi/src/Marconi/CardanoAPI.hs
Comment (eyeinsky) [inline marconi/src/Marconi/CardanoAPI.hs]: I think this isn't worth it: exporting TxIn from here and using it imported Marconi.CardanoAPI in other modules will give rise to the question of how is it different from the one in Cardano.Api, and the answer is that there are no differences. Maybe when we get the package `cardano-api-extras` that will serve as a front for cardano-api that we directly control and which' contents we want to upstream, it would make sense to re-export Cardano.Api types from there, but otherwise I think it's better to use things from their original location as it builds upon knowledge on having used those modules earlier.

Comment (berewt) [reply marconi/src/Marconi/CardanoAPI.hs]: I hesitated to do so for that exact reason. The pros were that we centralised our dependencies on cardano-api, whether or not these types are era-dependent. But you're right, even if reexporting is a common practice, it probably doesn't worth the effort here.

File: marconi/src/Marconi/Index/Utxo.hs
Comment (berewt) [reply marconi/src/Marconi/Index/Utxo.hs:188]: No problem, I have this bad habit of putting everything in one lens when I see one usage of them in an expression, I'm reverting it.

File: marconi/src/Marconi/CardanoAPI.hs
Comment (eyeinsky) [reply marconi/src/Marconi/CardanoAPI.hs]: Reexporting is fine for exporting definitions from modules that are otherwise internal, or to provide a nice API from a single location. But Cardano.Api is not internal, and it already is _the_ entrypoint to what cardano-api has to offer; and also as Marconi.CardanoAPI doesn't export all of Cardano.Api then most of the time Cardano.Api will already be imported, so having one type go through another module raises the question of why - does it do anything different? - but it doesn't, and it isn't era-dependent either.

**Review by eyeinsky — APPROVED** [review]
LGTM!

**Pull request merged**
